build-and-stage-preconfigured.ps1

```diff
@@ -45,4 +47,4 @@ foreach ($paramName in $allParams.Keys) {
 Write-Host ""
 Write-Host "Running build-and-stage.ps1..."
 Write-Host ""
-./build-and-stage.ps1 -PackageAndFeatures $packageAndFeatures -LinkType $linkType -BuildType $buildType -StagedArtifactsPath $StagedArtifactsPath -ReleaseTagBaseName $tagBaseName -PackageDisplayName $PackageDisplayName
+./build-and-stage.ps1 -PackageAndFeatures $packageAndFeatures -LinkType $linkType -BuildType $buildType -StagedArtifactsPath $StagedArtifactsPath -ReleaseTagBaseName $tagBaseName -PackageDisplayName $PackageDisplayName -VcpkgHash $vcpkgHash
```

Comment:
It looks like this is one of the few lines of code in this script that actually does something. On my first, ignorant, initial read-through, it's hard to differentiate between code that outputs variables and code that is actually doing something.

It would be nice if this function read something as simple as:

```
$packages  = Load-PackagesFromDisk()
if ( $packages doesn't have $packageToBuild ) {
    exit
}

Invoke-BuildAndStage -Package $packageToBuild
```

If we want to add debugging or status information, ideally that is confined to a single function.

Reply:
Yeah, it is kind of verbose and prints a lot of debugging info.  I'll pare that down a bunch.

Reply:
These have been greatly simplified:
- https://github.com/TechSmith/ThirdParty-Packages-vcpkg/blob/e3729cea9d6ca540731b243d618bdacdde5610a1/build-and-stage-preconfigured.ps1
- https://github.com/TechSmith/ThirdParty-Packages-vcpkg/blob/e3729cea9d6ca540731b243d618bdacdde5610a1/build-and-stage.ps1